src/main/java/com/iridium/iridiumskyblock/listeners/BlockExplodeListener.java

```diff
@@ -9,9 +11,13 @@
 
     @EventHandler(ignoreCancelled = true)
     public void onBlockExplode(BlockExplodeEvent event) {
-        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getBlock().getLocation()).ifPresent(island ->
-                event.blockList().removeIf(block -> !island.isInIsland(block.getLocation()))
-        );
+        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getBlock().getLocation()).ifPresent(island -> {
+            IslandSetting tntExplosion = IridiumSkyblock.getInstance().getIslandManager().getIslandSetting(island, SettingType.TNT_DAMAGE);
+            if (!tntExplosion.getBooleanValue()) {
+                event.setCancelled(true);
```

Comment:
You should return too
```suggestion
                event.setCancelled(true);
                return;
```